File: archive/c/c-sharp/SleepSort.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;

public class SleepSort
{
    public static void ErrorAndExit()
    {
        Console.WriteLine("Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"");
        Environment.Exit(1);
    }

    public static void Main(string[] args)
    {
        if (args.Length != 1)
            ErrorAndExit();
        try
        {
            var xs = args[0].Split(',').Select(i => Int32.Parse(i.Trim())).ToList();
            if (xs.Count <= 1)
                ErrorAndExit();

            Queue<int> sortedXs = new Queue<int>();

            Task.WaitAll(xs.Select(x =>
                Task.Delay(x * 1000).ContinueWith(_ => sortedXs.Enqueue(x))
            ).ToArray());

            Console.WriteLine(string.Join(", ", sortedXs));
        }
        catch
        {
            ErrorAndExit();
        }
    }
}

File: archive/v/visual-basic/sleep-sort.vb
Imports System.Collections.Concurrent

Public Module SleepSort

    Private Sub ShowUsage()
        Console.WriteLine("Usage: please provide a list of at least two integers to sort in the format ""1, 2, 3, 4, 5""")
        Environment.Exit(1)
    End Sub

    Public Sub Main(args As String())

        If args.Length <> 1 Then
            ShowUsage()
        End If

        Try
            Dim xs = args(0).
                Split(","c, StringSplitOptions.RemoveEmptyEntries).
                Select(Function(i) Integer.Parse(i.Trim())).
                ToList()

            If xs.Count <= 1 Then
                ShowUsage()
            End If

            Dim sortedXs As New ConcurrentQueue(Of Integer)
            Dim tasks As New List(Of Task)

            For Each x In xs

                Dim captured = x

                tasks.Add(Task.Run(Async Function()
                                       Await Task.Delay(captured * 1000)
                                       sortedXs.Enqueue(captured)
                                   End Function))

            Next

            Task.WaitAll(tasks.ToArray())

            Console.WriteLine(String.Join(", ", sortedXs))

        Catch
            ShowUsage()
        End Try

    End Sub

End Module